src/Algebra/Properties/Monoid/Reasoning.agda

```diff
@@ -0,0 +1,68 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Equational reasoning for monoids
+-- (Utilities for identity and cancellation reasoning, extending semigroup reasoning)
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Algebra.Bundles using (Monoid)
+
+module Algebra.Properties.Monoid.Reasoning {o ℓ} (M : Monoid o ℓ) where
+
+open Monoid M
+    using (Carrier; _∙_; _≈_; setoid; isMagma; semigroup; ε; sym; identityˡ
+    ; identityʳ ; ∙-cong; refl; assoc; ∙-congˡ; ∙-congʳ; trans)
+open import Relation.Binary.Reasoning.Setoid setoid
+open import Algebra.Properties.Semigroup.Reasoning semigroup public
+
+module Identity {a : Carrier } where
+    id-unique : (∀ b → b ∙ a ≈ b) → a ≈ ε
+    id-unique b∙a≈b = trans (sym (identityˡ a)) (b∙a≈b ε)
+
+    id-comm : a ∙ ε ≈ ε ∙ a
+    id-comm = trans (identityʳ a) (sym (identityˡ a))
+
+    id-comm-sym : ε ∙ a ≈ a ∙ ε
+    id-comm-sym = sym id-comm
```

Comment:
I now think that these explicit symmetric forms are excessive/superfluous.

Reply:
`id-comm-sym` (and `id-comm`) are used surprisingly often in `agda-categories`. But I think that might be an artifact of how classical category theory is "formalized" in textbooks that when *actually* formalized adds a ridiculous number of `id` that need to be shuffled around.

It also happens when doing naive `foldr` or naive partial evaluation: you get spurious `id`s, and sometimes you need to shuffle these around. So while I'm not claiming `id-comm-sym` would pay its weight here, there's a good reason why it is being proposed.

Reply:
Interesting, thanks for the insight. All of this seems only further evidence towards developing `Solver`-style/normal form equational reasoning as in #2629 ...